### ISSUE TITLE
False-positive "cannot move"

### ISSUE THREAD

**Issue by lovasoa** [intellij-rust/intellij-rust#4307]
## Environment

* **IntelliJ Rust plugin version:** 0.2.103.2131-192
* **Rust toolchain version:** 1.37.0 (eae3437df 2019-08-13) x86_64-apple-darwin
* **IDE name and version:** IC-192.6262.58
* **Operating system:** macOS 10.14.6

## Problem description
When using an `if let` binding with a slice without a reference, intellij-rust raises a false-positive cannot move warning.

<img width="454" alt="image" src="https://user-images.githubusercontent.com/552629/63647851-8cb78480-c727-11e9-9cc5-35f4e385aaef.png">

## Steps to reproduce

The following code compiles just fine but intellij-rust raises a warning:

```rust
fn f(v: Vec<i32>) {
    if let [a, b, c] = v[..] {
        println!("{}", a + b + c);
    }
}
```

**Comment by lovasoa** [comment]
I am not sure it's the same bug, but a similar situation is:

![image](https://user-images.githubusercontent.com/552629/63705957-9081fe80-c82e-11e9-91da-2466e0979527.png)


Should I open a separate issue for that ?

**Comment by ortem** [comment]
@lovasoa Yes, open a separate issue for mismatched types, please.

**Comment by lovasoa** [comment]
@ortem : I cannot reproduce the "mismatched types" issue anymore. May it be because I switched to the new macro expansion engine ?

**Comment by ortem** [comment]
@lovasoa You're right, enabling new macro expansion engine solves the problem! But I think you should open the issue anyway. The "mismatched types" issue should be closed when the new macro expansion engine is enabled by default.
